Broke out angular-cli docs and added required section about dependencies

pontusr opened the pull request:
It's looking like angular-cli and the ngrx router will be a common combination. I broke out the angular-cli section from the SystemJS file and also added what I believe is a neccessary step (for now) to `npm install´ the dependencies.

brandonroberts commented:
Thanks for the PR! There isn't a requirement to `npm install` the dependencies separately as `path-to-regexp` is a dependency of `ngrx/router` and it brings the other dependencies with it.

MikeRyan52 commented:
@brandonroberts I've been given to understand that there is a path-to-regexp conflict between ngrx/router and the CLI, requiring developers to explicitly install that dependency.

brandonroberts commented:
Yes, but `path-to-regexp` and `query-string` are the only explicit dependencies. We shouldn't need to add every sub-dependency also.

pontusr commented:
I also feel that installing the explicitly installing the dependencies is not the right way to go. However, I'm not well-versed with SystemJS/Broccoli/Angular CLI to pinpoint the root cause.

The documented step in the PR to install the dependencies using NPM solves the issue however.

Steps to repro the issue with combining angular-cli and ngrx/router according to current docs:

1. Install angular-cli according to https://github.com/angular/angular-cli/ :
```Bash
npm install -g angular-cli
```

2. Generate, build for dev and serve:
```Bash
ng new ngrxtest
cd ngrxtest
ng serve
```
Confirm app us building and working without errors.

3. Build for production and serve:
```Bash
ng serve -prod
```
Confirm app is building and working without errors

4. Install ngrx/router according to https://github.com/nxrx/router/ :
```Bash
npm install @ngrx/router @ngrx/core --save
```

5. Configure CLI project according to https://github.com/ngrx/router/blob/master/docs/overview/systemjs.md#angular-cli-configuration

6. Setup the routing according to the example on https://github.com/nxrx/router/

7. Build and serve dev:
```Bash
ng serve
```
Build works fine but browser console gives:
`zone.js:101 GET http://localhost:4200/vendor/query-string/index.js 404 (Not Found)`

8. Build for production and serve:
```Bash
ng serve -prod
```
Build error: `Error on fetch for vendor/query-string/index.js at file:///E:/Source/ngrxtest/tmp/bundle_plugin-input_base_path-C198oAaa.tmp/0/vendor/query-string/index.js`

Installing query-string using NPM will then render the same error for the next dependency until you have installed all dependencies using NPM - then it works.

brandonroberts commented:
@pontusr can you share a little bit about your environment? I'm not seeing the same issue on OSX 10.10 with node 5.5.0 and npm 3.5.3

brandonroberts commented on docs/overview/angular-cli.md:
````diff
@@ -0,0 +1,53 @@
+Install the dependencies into your CLI project via npm:
+
+```
+npm install path-to-regexp isarray query-string strict-uri-encode object-assign
````
Add `--save` to this line

brandonroberts commented on docs/overview/angular-cli.md:
````diff
@@ -0,0 +1,53 @@
+
+```js
+/** Map relative paths to URLs. */
+const map: any = {
````
Go ahead and clean up the quotes to make them all single quotes.

pontusr commented:
I had this issue on Windows 10, node 4.4.5, npm 2.15.5. After upgrading to node 6.2.1 with npm 3.9.3 I do not see the issue any more.

pontusr commented:
I fixed the quotes, removed the section about the installation of the npm dependencies as it doesn't seem to be needed and made the instructions more logical and complete (IMO).

pontusr commented:
Some further testing on another project makes me believe that there is something with regards to the dependencies that ngrx/router needs that require node >4.4.5 and/or npm >2.15.5 - at least on Windows 10. Further testing would be needed unless someon knows which explicit feature/issue in node/npm is required. After pinpointing the requirement this should probably go into readme.md.

brandonroberts commented:
Thanks for providing that information. It makes more sense why you would need to explicitly install those dependencies also if you're still on npm < 3.x. We'll look at adding a section to the readme for different npm versions.
